app/src/main/java/org/literacyapp/authentication/TrainingHelper.java

```diff
@@ -0,0 +1,96 @@
+import ch.zhaw.facerecognitionlibrary.Recognition.Recognition;
+import ch.zhaw.facerecognitionlibrary.Recognition.SupportVectorMachine;
+import ch.zhaw.facerecognitionlibrary.Recognition.TensorFlow;
+
+/**
+ * Created by sladomic on 26.11.16.
+ */
+
+public class TrainingHelper {
+    private Context context;
+    private DaoSession daoSession;
+    private StudentImageDao studentImageDao;
+    private StudentImageFeatureDao studentImageFeatureDao;
+    private SupportVectorMachine svm;
+
+    static {
+        if (!OpenCVLoader.initDebug()) {
+            // Handle initialization error
+        }
+    }
+
+    public TrainingHelper(Context context){
+        this.context = context;
+        LiteracyApplication literacyApplication = (LiteracyApplication) context.getApplicationContext();
+        daoSession = literacyApplication.getDaoSession();
+        studentImageDao = daoSession.getStudentImageDao();
+        studentImageFeatureDao = daoSession.getStudentImageFeatureDao();
+        svm = new SupportVectorMachine(context, Recognition.TRAINING);
+    }
+
+
+    public void extractFeatures(){
+        // Get all the StudentImages where the features haven't been extracted yet
+        List<StudentImage> studentImageList = studentImageDao.queryBuilder()
+                .where(StudentImageDao.Properties.StudentImageFeatureId.eq(0))
+                .list();
+        Log.d(getClass().getName(), "Number of StudentImages, where the features haven't been extracted yet: " + studentImageList.size());
+        TensorFlow tensorFlow = getInitializedTensorFlow();
+        for(StudentImage studentImage : studentImageList){
+            String svmVector = getSvmVector(tensorFlow, studentImage);
+            storeStudentImageFeature(studentImage, svmVector);
+        }
+    }
+
+    private void storeStudentImageFeature(StudentImage studentImage, String svmVector){
+        StudentImageFeature studentImageFeature = new StudentImageFeature(studentImage.getId(), Calendar.getInstance(), svmVector);
+        studentImage.setStudentImageFeature(studentImageFeature);
+        studentImageFeatureDao.insert(studentImageFeature);
+        studentImageDao.update(studentImage);
+        Log.d(getClass().getName(), "StudentImageFeature for StudentImage with Id " + studentImage.getId() + " has been extracted and stored.");
+    }
+
+    private String getSvmVector(TensorFlow tensorFlow, StudentImage studentImage){
+        // Load image into OpenCV Mat object
+        Mat img = Imgcodecs.imread(studentImage.getImageFileUrl());
+        // Extract features from TensorFlow model
+        Mat featureVector = tensorFlow.getFeatureVector(img);
+        // Convert featureVector to SVM string
+        return svm.getSvmString(featureVector, String.valueOf(studentImage.getStudentImageCollectionEvent().getStudentId()));
+    }
+
+    private TensorFlow getInitializedTensorFlow(){
+        String model = AiHelper.getModelDirectory() + "/vgg_faces.pb";
```

Comment:
Does this mean that we have to manually place the `vgg_faces.pb` on every device for it to work?